platform/macos/editor/embedded_process_macos.h

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include "editor/plugins/embedded_process.h"
+
+class DisplayServerMacOS;
+
+class LayerHost : public Control {
+	GDCLASS(LayerHost, Control);
+
+	ScriptEditorDebugger *script_debugger = nullptr;
+
+	virtual void gui_input(const Ref<InputEvent> &p_event) override;
+
+protected:
+	void _notification(int p_what);
+
+public:
+	void set_script_debugger(ScriptEditorDebugger *p_debugger) {
+		script_debugger = p_debugger;
+	}
+};
+
+class EmbeddedProcessMacOS final : public EmbeddedProcessBase {
+	GDCLASS(EmbeddedProcessMacOS, EmbeddedProcessBase);
+
+	enum class EmbeddingState {
+		IDLE,
+		IN_PROGRESS,
+		COMPLETED,
+		FAILED,
+	};
+
+	DisplayServerMacOS *ds = nullptr;
+	EmbeddingState embedding_state = EmbeddingState::IDLE;
+	uint32_t context_id = 0;
+	ScriptEditorDebugger *script_debugger = nullptr;
+	LayerHost *layer_host = nullptr;
+	OS::ProcessID current_process_id = 0;
+
+	// Embedded process state.
+
+	/// @brief The current mouse mode of the embedded process.
+	DisplayServer::MouseMode mouse_mode = DisplayServer::MOUSE_MODE_VISIBLE;
+
+	void _try_embed_process();
+	void update_embedded_process() const;
+	void _joy_connection_changed(const int p_index, const bool p_connected) const;
```

Comment:
```suggestion
	void _joy_connection_changed(int p_index, bool p_connected) const;
```
No `const` for simple types

Reply:
(only in parameters)